apps/emqx_dashboard/src/emqx_dashboard_audit.erl

```diff
@@ -0,0 +1,51 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2023 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-module(emqx_dashboard_audit).
+
+-include_lib("emqx/include/logger.hrl").
+%% API
+-export([log/1]).
+
+log(Meta0) ->
+    #{req_start := ReqStart, req_end := ReqEnd, code := Code, method := Method} = Meta0,
+    Duration = erlang:convert_time_unit(ReqEnd - ReqStart, native, millisecond),
+    Level = level(Method, Code, Duration),
+    Username = maps:get(username, Meta0, <<"">>),
+    Meta1 = maps:without([req_start, req_end], Meta0),
+    Meta2 = Meta1#{time => logger:timestamp(), duration_ms => Duration},
+    Meta = emqx_utils:redact(Meta2),
+    ?AUDIT(
+        Level,
+        "from_api",
+        Meta#{
```

Comment:
Security risk: Is it necessary to avoid printing the plain password from an HTTP request or HEADER?

Reply:
Sorry, it has processed in L#30